File: src/B_1260/B_1260.java
package B_1260;
import java.util.*;

public class B_1260 {
    public static void main(String[] arg){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int v = sc.nextInt();
        boolean[] visited = new boolean[n+1];
        boolean[] visited1 = new boolean[n+1];
        LinkedList<Integer>[] adjList = new LinkedList[n+1];
        for(int i=0; i<=n; i++){
            adjList[i] = new LinkedList<>();
        }
        for(int i=0; i<m; i++){
            int v1=sc.nextInt();
            int v2=sc.nextInt();
            adjList[v1].add(v2);
            adjList[v2].add(v1);
        }
        for(int i=0; i<=n; i++){
            Collections.sort(adjList[i]);
        }
        dfs_list(v,adjList,visited);
        System.out.println();
        bfs_list(v,adjList,visited1);


    }
    public static void bfs_list(int v, LinkedList<Integer>[] adjList, boolean[] visited){
        Queue<Integer> q= new LinkedList<>();
        visited[v]=true;
        q.add(v);
        while(q.size() != 0){
            v= q.poll();
            System.out.print(v+" ");
            for (int w : adjList[v]) {
                if (!visited[w]) {
                    visited[w] = true;
                    q.add(w);
                }
            }
        }

    }
    public static void dfs_list(int v, LinkedList<Integer>[] adjList,boolean[] visited){
        visited[v]=true;
        System.out.print(v+" ");
        for (int w : adjList[v]) {
            if (!visited[w]) dfs_list(w, adjList, visited);
        }
    }
}

File: src/B_1260/B_1260.kt
import java.util.*


    fun main(arg: Array<String>) {
        val sc = Scanner(System.`in`)
        val n = sc.nextInt()
        val m = sc.nextInt()
        val v = sc.nextInt()
        val visited = BooleanArray(n + 1)
        val visited1 = BooleanArray(n + 1)
        val adjList: Array<LinkedList<Int>?> = arrayOfNulls<LinkedList<Int>>(n + 1)
        for (i in 0..n) {
            adjList[i] = LinkedList()
        }
        for (i in 0 until m) {
            val v1 = sc.nextInt()
            val v2 = sc.nextInt()
            adjList[v1]!!.add(v2)
            adjList[v2]!!.add(v1)
        }
        for (i in 0..n) {
            Collections.sort(adjList[i])
        }
        dfs_list(v, adjList, visited)
        println("")
        bfs_list(v, adjList, visited1)
    }

    fun bfs_list(v: Int, adjList: Array<LinkedList<Int>?>, visited: BooleanArray) {
        var v = v
        val q: Queue<Int> = LinkedList()
        visited[v] = true
        q.add(v)
        while (q.size != 0) {
            v = q.poll()
            print("$v ")
            val iter: Iterator<Int> = adjList[v]!!.listIterator()
            while (iter.hasNext()) {
                val w = iter.next()
                if (!visited[w]) {
                    visited[w] = true
                    q.add(w)
                }
            }
        }
    }

    fun dfs_list(v: Int, adjList: Array<LinkedList<Int>?>, visited: BooleanArray) {
        visited[v] = true
        print("$v ")
        val iter: Iterator<Int> = adjList[v]!!.listIterator()
        while (iter.hasNext()) {
            val w = iter.next()
            if (!visited[w]) dfs_list(w, adjList, visited)
        }
    }

